diff -ur jexcelapi-2.6.5/src/jxl/DateCell.java jexcelapi-2.6.5-patched-dateCellNumber/src/jxl/DateCell.java
--- jexcelapi-2.6.5/src/jxl/DateCell.java	2007-08-22 14:13:54.000000000 +0300
+++ jexcelapi-2.6.5-patched-dateCellNumber/src/jxl/DateCell.java	2007-08-22 16:53:12.000000000 +0300
@@ -35,6 +35,13 @@
   public Date getDate();
 
   /**
+   * Gets the contents for this cell as a number.
+   *
+   * @return the cell contents
+   */
+  public double getValue();
+
+  /**
    * Indicates whether the date value contained in this cell refers to a date,
    * or merely a time
    *
diff -ur jexcelapi-2.6.5/src/jxl/read/biff/DateFormulaRecord.java jexcelapi-2.6.5-patched-dateCellNumber/src/jxl/read/biff/DateFormulaRecord.java
--- jexcelapi-2.6.5/src/jxl/read/biff/DateFormulaRecord.java	2007-08-22 14:13:53.000000000 +0300
+++ jexcelapi-2.6.5-patched-dateCellNumber/src/jxl/read/biff/DateFormulaRecord.java	2007-08-22 16:53:12.000000000 +0300
@@ -130,16 +130,6 @@
   }
 
   /**
-   * Interface method which returns the value
-   *
-   * @return the last calculated value of the formula
-   */
-  public double getValue()
-  {
-    return 0;
-  }
-
-  /**
    * Dummy implementation in order to adhere to the NumberCell interface
    *
    * @return NULL
diff -ur jexcelapi-2.6.5/src/jxl/read/biff/DateRecord.java jexcelapi-2.6.5-patched-dateCellNumber/src/jxl/read/biff/DateRecord.java
--- jexcelapi-2.6.5/src/jxl/read/biff/DateRecord.java	2007-08-22 14:13:53.000000000 +0300
+++ jexcelapi-2.6.5-patched-dateCellNumber/src/jxl/read/biff/DateRecord.java	2007-08-22 16:53:12.000000000 +0300
@@ -45,6 +45,10 @@
   private static Logger logger  = Logger.getLogger(DateRecord.class);
 
   /**
+   * The excel value of the date
+   */
+  private double value;
+  /**
    * The date represented within this cell
    */
   private Date date;
@@ -143,6 +147,7 @@
   {
     row = num.getRow();
     column = num.getColumn();
+    value = num.getValue();
     xfIndex = xfi;
     formattingRecords = fr;
     sheet = si;
@@ -226,6 +231,16 @@
   }
 
   /**
+   * Gets the contents for this cell as a number
+   *
+   * @return the cell contents
+   */
+  public double getValue()
+  {
+    return value;
+  }
+
+  /**
    * Gets the cell contents as a string.  This method will use the java
    * equivalent of the excel formatting string
    *
diff -ur jexcelapi-2.6.5/src/jxl/write/biff/DateRecord.java jexcelapi-2.6.5-patched-dateCellNumber/src/jxl/write/biff/DateRecord.java
--- jexcelapi-2.6.5/src/jxl/write/biff/DateRecord.java	2007-08-22 14:13:53.000000000 +0300
+++ jexcelapi-2.6.5-patched-dateCellNumber/src/jxl/write/biff/DateRecord.java	2007-08-22 16:53:12.000000000 +0300
@@ -36,7 +36,7 @@
 /**
  * A date stored in the database
  */
-public abstract class DateRecord extends CellValue
+public abstract class DateRecord extends CellValue implements DateCell
 {
   /**
    * The logger
@@ -314,6 +314,16 @@
   }
 
   /**
+   * Gets the contents for this cell as a number
+   *
+   * @return the cell contents
+   */
+  public double getValue()
+  {
+    return value;
+  }
+
+  /**
    * Indicates whether the date value contained in this cell refers to a date,
    * or merely a time.  When writing a cell, all dates are fully defined,
    * even if they refer to a time
diff -ur jexcelapi-2.6.5/src/jxl/write/biff/ReadDateFormulaRecord.java jexcelapi-2.6.5-patched-dateCellNumber/src/jxl/write/biff/ReadDateFormulaRecord.java
--- jexcelapi-2.6.5/src/jxl/write/biff/ReadDateFormulaRecord.java	2007-08-22 14:13:53.000000000 +0300
+++ jexcelapi-2.6.5-patched-dateCellNumber/src/jxl/write/biff/ReadDateFormulaRecord.java	2007-08-22 16:53:12.000000000 +0300
@@ -52,6 +52,16 @@
   }
 
   /**
+   * Gets the contents for this cell as a number.
+   *
+   * @return the cell contents
+   */
+  public double getValue()
+  {
+    return ( (DateFormulaCell) getReadFormula()).getValue();
+  }
+
+  /**
    * Indicates whether the date value contained in this cell refers to a date,
    * or merely a time
    *
