diff -r af79210332a9 -r 72c0c5915032 src/jxl/biff/formula/Attribute.java
--- a/src/jxl/biff/formula/Attribute.java	Mon Aug 20 11:08:33 2007 +0200
+++ b/src/jxl/biff/formula/Attribute.java	Mon Aug 20 14:59:00 2007 +0200
@@ -54,6 +54,7 @@ class Attribute extends Operator impleme
 
   private static final int SUM_MASK  = 0x10;
   private static final int IF_MASK   = 0x02;
+  private static final int CHOOSE_MASK = 0x04;
   private static final int GOTO_MASK = 0x08;
 
   /**
@@ -116,7 +117,11 @@ class Attribute extends Operator impleme
   {
     options = data[pos];
     word = IntegerHelper.getInt(data[pos + 1], data[pos + 2]);
-    return 3;
+    if (!isChoose()) return 3;
+    // See http://sc.openoffice.org/excelfileformat.pdf, page 69
+    // "word" contains the number of jumps by index.
+    // And there is an additional final jump to the choose function itself.
+    return 3 + (word + 1) * 2;
   }
 
   /**
@@ -147,6 +152,16 @@ class Attribute extends Operator impleme
   public boolean isIf()
   {
     return (options & IF_MASK) != 0;
+  }
+
+  /**
+   * Queries whether this attribute is a CHOOSE
+   *
+   * @return TRUE if this is a CHOOSE, FALSE otherwise
+   */
+  public boolean isChoose()
+  {
+    return (options & CHOOSE_MASK) != 0;
   }
 
   /**

