diff -ur jexcelapi-2.6.3/src/jxl/biff/formula/Function.java jexcelapi-2.6.3-patched/src/jxl/biff/formula/Function.java
--- jexcelapi-2.6.3/src/jxl/biff/formula/Function.java	2006-09-21 15:55:32.000000000 +0300
+++ jexcelapi-2.6.3-patched/src/jxl/biff/formula/Function.java	2007-04-05 15:39:25.000000000 +0300
@@ -293,8 +293,10 @@
     new Function(0x3b, "pmt", 0xff);
   public static final Function RATE =
     new Function(0x3c, "rate", 0xff);
-  //public static final Function MIRR =  new Function(0x3d, "MIRR",);
-  //public static final Function IRR =  new Function(0x3e, "IRR",);
+  public static final Function MIRR =
+    new Function(0x3d, "mirr", 3);
+  public static final Function IRR =
+    new Function(0x3e, "irr", 2);
   public static final Function RAND =
     new Function(0x3f, "rand", 0);
   public static final Function MATCH =
@@ -349,7 +351,7 @@
   //public static final Function SELECTION =  new Function(0x5f,"SELECTION",);
   //public static final Function RESULT =  new Function(0x60, "RESULT",);
   public static final Function ATAN2  =
-    new Function(0x61, "atan2", 1);
+    new Function(0x61, "atan2", 2);
   public static final Function ASIN =
     new Function(0x62, "asin", 1);
   public static final Function ACOS =
@@ -425,7 +427,7 @@
   public static final Function SLN =
     new Function(0x8e, "sln", 3);
   public static final Function SYD =
-    new Function(0x8f, "syd", 3);
+    new Function(0x8f, "syd", 4);
   public static final Function DDB =
     new Function(0x90, "ddb", 0xff);
   //public static final Function GETDEF =  new Function(0x91, "GETDEF",);
@@ -545,6 +547,8 @@
     new Function(0xea, "atanh", 1);
   public static final Function INFO =
     new Function(0xf4, "info", 1);
+  public static final Function DB =
+	new Function(0xf7, "db", 0xff);
   public static final Function AVEDEV =
     new Function(0x10d, "avedev", 0XFF);
   public static final Function BETADIST =
