diff -ur jexcelapi-2.6.4/src/jxl/write/biff/WritableWorkbookImpl.java jexcelapi-2.6.4-patched/src/jxl/write/biff/WritableWorkbookImpl.java
--- jexcelapi-2.6.4/src/jxl/write/biff/WritableWorkbookImpl.java	2007-05-29 13:04:19.000000000 +0300
+++ jexcelapi-2.6.4-patched/src/jxl/write/biff/WritableWorkbookImpl.java	2007-05-29 13:06:55.000000000 +0300
@@ -1723,6 +1723,15 @@
   }  
   
   /**
+   * Remove all area names from this workbook.
+   */
+  public void removeAllAreaNames()
+  {
+    nameRecords.clear();
+    names = null;
+  }
+
+  /**
    * Accessor for the workbook settings
    */
   WorkbookSettings getSettings()
diff -ur jexcelapi-2.6.4/src/jxl/write/WritableWorkbook.java jexcelapi-2.6.4-patched/src/jxl/write/WritableWorkbook.java
--- jexcelapi-2.6.4/src/jxl/write/WritableWorkbook.java	2007-05-29 13:04:19.000000000 +0300
+++ jexcelapi-2.6.4-patched/src/jxl/write/WritableWorkbook.java	2007-05-29 13:06:55.000000000 +0300
@@ -296,6 +296,11 @@
                                    int lastRow);
 
   /**
+   * Remove all area names from this workbook.
+   */
+  public abstract void removeAllAreaNames();
+
+  /**
    * Sets a new output file.  This allows the same workbook to be
    * written to various different output files without having to
    * read in any templates again
